### DISCUSSION TITLE
Pre-compile svg icon assets using `vector_graphics_compiler`

### DISCUSSION THREAD

**Discussion by aebrahim** [firebase/FirebaseUI-Flutter#240] [Ideas]
The social login icons are packaged as strings containing the raw svg, which are then rendered using `flutter_svg`. However, under the hood, using `flutter_svg` means also including its dependency `vector_graphics_compiler` with the application, which at runtime converts raw svg to a compiled `VectorGraphic` to display.

Given that the svg string is known at build time, it is possible to pre-compile it as a `VectorGraphic` at build time. Advantages:
1. Would no longer pull in the `vector_graphics_compiler` dependency, which adds ~180 KB to the build.
2. Slightly more efficient runtime as the precompiled `VectorGraphic` can be a const.

**Comment by lesnitsky** [comment]
Great suggestion, I will be looking into implementing this. Will be tracked under #247
